hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/keyvalue/statemachine/background/BlockDeletingService.java

```diff
@@ -433,23 +457,30 @@ public ContainerBackgroundTaskResult deleteViaSchema2(
       }
     }
 
-    private void deleteTransactions(List<DeletedBlocksTransaction> delBlocks,
+    private int deleteTransactions(List<DeletedBlocksTransaction> delBlocks,
         Handler handler, Table<String, BlockData> blockDataTable,
-        Container container) throws IOException {
+        Container container)
+        throws IOException {
+      int blocksDeleted = 0;
       for (DeletedBlocksTransaction entry : delBlocks) {
         for (Long blkLong : entry.getLocalIDList()) {
           String blk = blkLong.toString();
           BlockData blkInfo = blockDataTable.get(blk);
           LOG.debug("Deleting block {}", blk);
           try {
             handler.deleteBlock(container, blkInfo);
+            blocksDeleted++;
+            if (blocksDeleted == blockDeletionLimitPerInterval) {
+              return blocksDeleted;
+            }
```

Comment:
We can remove this change. We should either delete every block in a transaction or not process that transaction.

Reply:
Removed